File: go/eu013.go
// Project Euler problem 13
// Solution by Kevin Retzke, June 2012
package main

import (
	"bufio"
	"fmt"
	"io"
	"math/big"
	"os"
)

func main() {
	f, err := os.Open("../data/eu013.dat")
	if err != nil {
		os.Exit(1)
	}
	defer f.Close()
	r := bufio.NewReader(f)

	sum := big.NewInt(0)

	for {
		line, err := r.ReadString('\n')
		if err != nil {
			if err == io.EOF {
				break
			}
			os.Exit(1)
		}
		n := new(big.Int)
		n.SetString(line, 10)
		sum.Add(sum, n)
	}
	fmt.Println(sum.String()[0:10])
}

File: rust/src/bin/eu013.rs
extern crate pe;

use pe::BigInt;

use std::io;
use std::io::Read;
use std::fs::File;


fn sum_numbers_from_file(filename:&str) -> Result<BigInt, io::Error> {
    let mut f = File::open(filename)?;
    let mut s = String::new();
    f.read_to_string(&mut s)?;
    let mut n = BigInt::new();
    for line in s.lines() {
        let m = BigInt::new_from_string(line);
        //println!("{} {}",n,m);
        n.add_in_place(&m);
    }
    Ok(n)
}


fn main() {
    match sum_numbers_from_file("../data/eu013.dat") {
        Ok(n) => println!("result: {}",&n.to_string()[0..10]),
        Err(e) => println!("error: {}",e),
    }
}
